### DISCUSSION TITLE
PRACH issue with FDD oRU

### DISCUSSION THREAD

**Discussion by wei-lou** [srsran/srsRAN_Project#524] [General Help] [answered]
I am trying to integrate the srsRAN 23.10 with FDD n71 ORU. The RU supports PRACH format 0 and 2. I picked the format 0, PRACH configuration Index 7. Rest configurations are in the attached file. When I started gnb, it indicated some issues with the PRACH configuration.  Also from pcap, it didn't captured any PRACH packets. What are the issues here? Thanks,

```
labuser@srsgnb03:~$ sudo gnb -c srsRAN_Project/configs/FUJI_GEN1TB_RU_n71_10mhz_4x4.yml
[sudo] password for labuser:

The PRACH detector will not meet the performance requirements with the configuration {Format 0, ZCZ 0, SCS 1.25kHz, Rx ports 1}.

--== srsRAN gNB (commit 0b2702cca) ==--

Connecting to AMF on 192.168.70.210:38412
Initializing the Open FrontHaul Interface for sector#0: ul_compr=[BFP,9], dl_compr=[BFP,9], prach_compr=[BFP,9] prach_cp_enabled=false, downlink_broadcast=false.
Warning: Configured PRACH occasion collides with PUCCH RBs ([0..1) intersects [0..6)). Some interference between PUCCH and PRACH is expected.
Warning: Configured PRACH occasion collides with PUCCH RBs ([0..1) intersects [0..6)). Some interference between PUCCH and PRACH is expected.
Cell pci=140, bw=10 MHz, dl_arfcn=124400 (n71), dl_freq=622.0 MHz, dl_ssb_arfcn=123890, ul_freq=668.0 MHz

==== gNodeB started ===
Type <t> to view trace
q
Stopping ..

``` 
[gnb_prach_run01.log](https://github.com/srsran/srsRAN_Project/files/14705374/gnb_prach_run01.log)
[FUJI_GEN1TB_RU_n71_10mhz_4x4.yml.txt](https://github.com/srsran/srsRAN_Project/files/14705377/FUJI_GEN1TB_R
[gnb_prach_run01.pcap.gz](https://github.com/srsran/srsRAN_Project/files/14705378/gnb_prach_run01.pcap.gz)
U_n71_10mhz_4x4.yml.txt)

**Comment by wei-lou** [comment]
[FUJI_GEN1TB_RU_n71_10mhz_4x4.txt](https://github.com/srsran/srsRAN_Project/files/14705431/FUJI_GEN1TB_RU_n71_10mhz_4x4.txt)

**Comment by andrepuschmann** [comment]
If the RU support PRACH Cplane commands you should set the config to `true`. Also if the RU supports 10MHz natively you can uncomment/remove the `ru_bandwidth_MHz` option.
```
  #ru_bandwidth_MHz: 10
  is_prach_cp_enabled: true
```

**Reply by wei-lou** [reply]
Thanks, Andre! I commented out the "ru_bandwidth_MHZ: 10" line and changed the "is_prach_cp_enabled: true". I am still seeing the same issue, i.e. runtime warning and pcap has no prach packets.

```
labuser@srsgnb03:~$ sudo gnb -c srsRAN_Project/configs/FUJI_GEN1TB_RU_n71_10mhz_4x4.yml

The PRACH detector will not meet the performance requirements with the configuration {Format 0, ZCZ 0, SCS 1.25kHz, Rx ports 1}.

--== srsRAN gNB (commit 0b2702cca) ==--

Connecting to AMF on 192.168.70.210:38412
Initializing the Open FrontHaul Interface for sector#0: ul_compr=[BFP,9], dl_compr=[BFP,9], prach_compr=[BFP,9] prach_cp_enabled=true, downlink_broadcast=false.
Warning: Configured PRACH occasion collides with PUCCH RBs ([0..1) intersects [0..6)). Some interference between PUCCH and PRACH is expected.
Warning: Configured PRACH occasion collides with PUCCH RBs ([0..1) intersects [0..6)). Some interference between PUCCH and PRACH is expected.
Cell pci=140, bw=10 MHz, dl_arfcn=124400 (n71), dl_freq=622.0 MHz, dl_ssb_arfcn=123890, ul_freq=668.0 MHz

==== gNodeB started ===
Type <t> to view trace
q
Stopping ..
[gnb_prach_run03.pcap.gz](https://github.com/srsran/srsRAN_Project/files/14706974/gnb_prach_run03.pcap.gz)
[gnb_prach_run03.log](https://github.com/srsran/srsRAN_Project/files/14706975/gnb_prach_run03.log)

```

**Reply by wei-lou** [reply]
Oh, logs did get uploaded. Here are the logs ... 
[gnb_prach_run03.log](https://github.com/srsran/srsRAN_Project/files/14706983/gnb_prach_run03.log)
[gnb_prach_run03.pcap.gz](https://github.com/srsran/srsRAN_Project/files/14706984/gnb_prach_run03.pcap.gz)

**Reply by andrepuschmann** [reply]
I am not seeing any traffic from the RU flowing to the DU. Is the RU indicating it is sending something at all? As for the PRACH Cplane we might have to revisit the FDD case again. As you know it's not a tested configuration.

**Comment by wei-lou** [comment]
a few steps forward ... now UE decoded the MIB and SIBs, but initial RACH failed at msg 2. There are many too late packages in the gnb.log for PRACH uplink. I am not sure this is related to the UL t1a related timings or not.
[srs_gnb_rach_run12.log](https://github.com/srsran/srsRAN_Project/files/14781401/srs_gnb_rach_run12.log)
[FUJI_GEN1TB_RU_n71_10mhz_4x4.txt](https://github.com/srsran/srsRAN_Project/files/14781406/FUJI_GEN1TB_RU_n71_10mhz_4x4.txt)
[srs_gen1_rach_run12.pcap.gz.zip](https://github.com/srsran/srsRAN_Project/files/14781411/srs_gen1_rach_run12.pcap.gz.zip)

**Comment by andrepuschmann** [comment]
UL OFH traffic from the RU is still missing.

**Reply by wei-lou** [reply]
you are right. I found some errors on RU processing-elements configuration which impacted the cu-vlan settings. Thanks

**Reply by wei-lou** [reply]
After corrected some configurations on the RU side, i have installed the latest version of srs software. Now I am getting error while trying to execute the code. What exactly the changes do I need to make to fix the issue?
```
sulabuser@srsgnb03:~$ sudo gnb -c srsRAN_Project/configs/FUJI_GEN1TB_RU_n71_10mhz_4x4.yml
[sudo] password for sulabuser:

The PRACH detector will not meet the performance requirements with the configuration {Format 0, ZCZ 0, SCS 1.25kHz, Rx ports 1}.
Transmission timing window length for DL Control-Plane must be bigger than the symbol duration (71.43us).
srsGNB ERROR: Invalid configuration detected.

``` 
My configuration file is as this.
[FUJI_GEN1TB_RU_n71_10mhz_4x4.txt](https://github.com/srsran/srsRAN_Project/files/14859432/FUJI_GEN1TB_RU_n71_10mhz_4x4.txt)

**Comment by dvdgrgrtt** [comment]
Hi @wei-lou,
as for the message "The PRACH detector will not meet the performance requirements with the configuration {Format 0, ZCZ 0, SCS 1.25kHz, Rx ports 1}.", don't pay too much attention to it. It simply means that with your configuration, the PRACH detector will probably not achieve the performance goals suggested by the standard (probability of detection > 99% and probability of false alarm < 0.1%). In most cases, however, it should be close enough and you should not see significant differences.

The problem is that there are too many PRACH configurations for us to fine-tune them all.

Hope this helps!

**Reply by andrepuschmann** [reply]
As for the `Transmission timing window length for DL Control-Plane must be bigger than the symbol duration (71.43us).
srsGNB ERROR: Invalid configuration detected.` you need to make sure the gap between min and max values for the DL are sufficiently big enough, i.e. tune `t1a_max_cp_dl` and `t1a_min_cp_dl`, i.e. by lowering the min value.

**Reply by wei-lou** [reply]
I re-installed the server OS and srs 23.10.1. Now no matter what timing values I give to it, on RU almost all DL packets are "arrived too late". The timing set05 was a working IOT values with other vendor's DU. 

![image](https://github.com/srsran/srsRAN_Project/assets/137309708/45ac2691-1227-446c-af36-f5b368132559)

**Reply by pgawlowicz** [reply]
@wei-lou any update on this issue?

**Reply by wei-lou** [reply, marked as answer]
Sorry for the late reply. I have been out of town and had no access to my computer. Actually the issue was resolved some time ago. Two things have been done on it. 1) Installed latest 24.04 pkg. 2) compiled the package with TA = 0 for our setup.
